libraries/AP_Mount/AP_Mount_Topotek.h

```diff
@@ -0,0 +1,242 @@
+
+    // request gimbal tracking status
+    void request_track_status(void);
+
+    // request gimbal basic information
+    void request_gimbal_basic_info(void);
+
+    // set angle target in radians
+    void send_angle_target(float roll_rad, float pitch_rad, float yaw_rad, bool yaw_is_ef);
+
+    // sets rate target in rad/s
+    void send_rate_target(float roll_rads, float pitch_rads, float yaw_rads, bool yaw_is_ef);
+
+    // send time and date to gimbal
+    bool send_time_to_gimbal();
+
+    // send GPS-related information to the gimbal
+    bool send_location_info(void);
+
+    // analysis the data information received
+    void analyse_packet_data(void);
+
+    // attitude information analysis of gimbal
+    void gimbal_angle_analyse(void);
+
+    // gimbal video information analysis
+    void gimbal_record_analyse(void);
+
+    // information analysis of gimbal storage card
+    void gimbal_sdcard_analyse(void);
+
+    // gimbal tracking information analysis
+    void gimbal_track_analyse(void);
+
+    // gimbal basic information analysis
+    void gimbal_basic_info_analyse(void);
+
+    // gimbal distance information analysis
+    void gimbal_dist_info_analyse(void);
+
+    // add check
+    void add_check(int8_t *cmd, uint8_t len);
+
+    // calculate checksum
+    int8_t calculate_crc(int8_t *cmd, uint8_t len);
+
+    // hexadecimal to character conversion
+    int8_t hex2char(int8_t data);
+
+    // character to number conversion
+    uint8_t char_to_number(int8_t data);
+
+    // send packet to gimbal
+    // returns true on success, false if serial port initialization failed
+    bool send_packet(int8_t* data_packet, uint8_t data_packet_len);
+
+    // set gimbal's lock vs follow mode
+    // lock should be true if gimbal should maintain an earth-frame target
+    // lock is false to follow / maintain a body-frame target
+    bool set_gimbal_lock(bool lock);
+
+    // uart connected to gimbal
+    AP_HAL::UARTDriver *_uart;
+
+    // members
+    bool _initialised;                                          // true once the serial port has been initialised
+    bool _recording;                                            // recording status (received from gimbal)
+    bool _is_tracking;                                          // whether to enable the tracking state
+    bool _sdcard_status;                                        // memory card status (received from gimbal)
+    bool _last_lock;                                            // last lock mode sent to gimbal
+    bool _got_gimbal_basic_info;                                // whether the gimbal information is obtained
+    bool _last_zoom_stop = false;                               // record the latest zoom stopped state.
+    bool _last_focus_stop = false;                              // record the latest focus stopped state.
+    uint8_t _sent_time_count = 0;                               // count of current time messages sent to gimbal
```

Comment:
why init?

Reply:
thanks.  all initialisation of members to zero have been removed.